### PR TITLE
Change from assay_name to assay.type

### LINKED ISSUE #189: Argument naming: "assay_name" (scater) vs. "assay.type" (scuttle)
We recently changed the argument name from `exprs_values` to `assay_name` in _scater_ in PR #187 . This is in sync with mia package family, which uses `assay_name` as well. 

The _scuttle_ package is also using `exprs_values`, and I proposed a similar change in [scuttle issue #21](https://github.com/LTLA/scuttle/issues/21#issuecomment-1416688770). The maintainer has decided to prefer dot arguments (`assay.type`) instead, and has aliases for past underscore arguments to keep the code functional.

That would be a bigger change as it would affect other arguments, too. I would be ready to consider that for mia/miaViz/miaTime/miaSim etc. packages in order to align the conventions. But that would seem to make sense only if both _scater_ and _scuttle_ would converge to the same. If these two packages will have different argument names, we will probably keep mia* as is, and consider additional aliases to support interoperability with both.

Just wanted to cross-check this before proceeding. Is moving to dot arguments something that the _scater_ package could consider?

### PR DESCRIPTION
**Pull request by antagomir**
This should fix #189 as discussed.

The chance concerns only the arguments containing `assay_name`, changing these argument names from underscore type to dot type.

Changing all other argument names in this way, or reviewing other argument names where this would support interoperability with _scuttle_ would be a heavier undertaking, perhaps best dealt with if specific needs will arise. Otherwise that will be mainly a stylistic issue.

### CODE REVIEW

**Comment by antagomir** [comment]
The latest commit adds the Deprecation messages.

Meanwhile I noticed that at least the following arguments are shared with or passed to _scuttle_, which has already moved to using dot names for arguments and deprecated these versions. I could add a similar change for these arguments in the same or another PR if you agree. It seems that the change should be done at the latest when _scuttle_ removes these arguments from deprecation stage

- `check_names` (perFeatureDF arguments; within the `ggfeatures` function I already updated to call check.names for the scuttle call)
- `use_altexps`
- `prefix_altexps`
- `use_dimred` -> this is dimred in _scater_ and  use.dimred in _scuttle_

**Comment by alanocallaghan** [comment]
Can you revert to [6c30bee](https://github.com/Alanocallaghan/scater/pull/194/commits/6c30beee89b1f23a2201a77d463eccd4722f77c3)? I don't want to add deprecation messages within this cycle.

**Comment by alanocallaghan** [comment]
Also if you were adding deprecation messages the message should only pop up when somebody actually uses the old argument name. As-is people would get a deprecation message even if they were using the new argument name or not using either.

Something like

```r
if (!missing(exprs_values)) {
  .Deprecated(msg="message")
}
```

**Comment by antagomir** [comment]
Ah I got it wrong when you gave the small comment that that the exprs_values is not deprecated. I thought it meant that the deprecation messages should be added.

I have now reverted to where it was before that comment.

I agree about the check before deprecation message, should have noticed that.

The question about the other arguments remains. I have listed above other arguments that scuttle has already deprecated (with message), and which are called in scater with their old names. It is possible to add those in this PR, or another new PR.

Is there something that still specifically has to be changed for this PR?

**Comment by alanocallaghan** [comment]
I haven't checked over the scuttle stuff yet, I think this PR is fine - if I want to change the other args I can do that myself anyways

**Comment by antagomir** [comment]
Ok great. Just let me know if anything still needs to be changed before merging. The checks seems to pass at least.

**Comment by antagomir** [comment]
The PR is ready from my side at least.

File: R/plotExpression.R [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
 #' @param shape_by Specification of a column metadata field or a feature to shape by, see the \code{by} argument in \code{?\link{retrieveCellInfo}} for possible values. 
 #' @param size_by Specification of a column metadata field or a feature to size by, see the \code{by} argument in \code{?\link{retrieveCellInfo}} for possible values. 
 #' @param order_by Specification of a column metadata field or a feature to order points by, see the \code{by} argument in \code{?\link{retrieveCellInfo}} for possible values.
-#' @param by_assay_name A string or integer scalar specifying which assay to obtain expression values from, 
-#' for use in point aesthetics - see the \code{assay_name} argument in \code{?\link{retrieveCellInfo}}. Also the alias \code{by_assay_name} is accepted.
+#' @param by.assay.type A string or integer scalar specifying which assay to obtain expression values from, 
+#' for use in point aesthetics - see the \code{assay.type} argument in \code{?\link{retrieveCellInfo}}. Also the alias \code{by.assay.type} is accepted.
```
Comment (alanocallaghan) [inline R/plotExpression.R]: `by.assay.type` can't be an alias for `by.assay.type`?

File: R/plotExpression.R [hunk 2]
```diff
@@ -6,14 +6,14 @@
 #' @param features A character vector or a list specifying the features to plot.
 #' If a list is supplied, each entry of the list can be a string, an AsIs-wrapped vector or a data.frame - see \code{?\link{retrieveCellInfo}}.
 #' @param x Specification of a column metadata field or a feature to show on the x-axis, see the \code{by} argument in \code{?\link{retrieveCellInfo}} for possible values. 
-#' @param assay_name A string or integer scalar specifying which assay in \code{assays(object)} to obtain expression values from. Also the alias \code{assay_name} is accepted.
+#' @param assay.type A string or integer scalar specifying which assay in \code{assays(object)} to obtain expression values from. Also the alias \code{assay.type} is accepted.
```
Comment (alanocallaghan) [inline R/plotExpression.R]: `assay.type` isn't alias for itself

**Comment by alanocallaghan** [comment]
Can you see my review comments this time?

**Review by alanocallaghan — CHANGES_REQUESTED** [review]
Bump the date and version, and fix the recursive alias

File: R/plotExpression.R (hunk 1 above)
Comment (antagomir) [reply R/plotExpression.R]: Fixed now in the PR update (coming).

File: R/plotExpression.R (hunk 2 above)
Comment (antagomir) [reply R/plotExpression.R]: done

**Review by alanocallaghan — APPROVED** [review]

**Comment by antagomir** [comment]
I did not receive any notification on the review comments but now noticed them. I have added these in the latest push, hopefully all ok but if not then let's continue to finalize.

**Comment by alanocallaghan** [comment]
Thanks! Sorry for the delay in looking at it

**Comment by antagomir** [comment]
I would be looking fwd to having updates pushed to Bioconductor, then we could adopt these in downstream packages and tutorials.

**Comment by alanocallaghan** [comment]
Devel branch is pushed to bioc

**Comment by antagomir** [comment]
Awesome - thanks!

**Pull request merged**